### PR TITLE
Module3pt2

### PR DESCRIPTION
**Pull request by JoLeeSoftwire**
fixing unit tests to pass 24hrs a day

### CODE REVIEW

File: src/tests/ViewModel_test.py [hunk 1]
```diff
@@ -4,31 +4,29 @@
...
-def test_correct_recent_done_items():
+def test_correct_recent_done_items(monkeypatch):
     # Given
-    task1DoneAt = datetime.now() - timedelta(minutes=15)
-    task2DoneAt = datetime.now() - timedelta(minutes=10)
-    task3DoneAt = datetime.strptime("2020-01-08T08:15:27.243860Z", dateformat)
-    task4DoneAt = datetime.strptime("2020-06-29T09:17:20.141163Z", dateformat)
-    task5DoneAt = datetime.now() - timedelta(minutes=1)
-    task6DoneAt = datetime.now() - timedelta(minutes=120)
+    monkeypatch.setattr(Task, "touchedOn", lambda self, date: self.last_modified == dummyToday)
```
Comment (hugh-emerson) [inline src/tests/ViewModel_test.py]: Your change seems a bit confusing/overcomplicated.

You could have just used the real function but set today/notToday equal to `datetime.now()` and `datetime.now() - timedelta(days=1)`

If your reasoning was that unit tests should mock other classes' functions, this is going about it a bit wrong. You aren't checking that touchedOn is used correctly (the date param is totally ignored in your monkeypatch function) and by using dummyToday/dummyNotToday, it's not immediately obvious to what extent the date matters in the test. I would keep it as `touched_today`, and just mock it to return true/false based on the id or something. 
But I don't think mocking is necessary I don't think is necessary for a little function like Task.touchedToday. Task is still basically just a data class.

I would say in general you should just use the real datetime and if there are problems, you should mock the current time (freezegun does it for Python). If you wanted to test a variety of times like before this PR, then you could just get the real date and set the time to 1 am, 11pm etc.

File: src/todolist/Task.py [hunk 2]
```diff
@@ -27,8 +27,8 @@ def from_trello(cls, item, status=Status.ToDo):
         task.last_modified = datetime.strptime(item['dateLastActivity'], dateformat)
         return task
 
-    def touchedToday(self):
-        return self.last_modified.date() == datetime.now().date()
+    def touchedOn(self, date):
```
Comment (hugh-emerson) [inline src/todolist/Task.py]: This should be reverted. If you are adding function parameters for testing reasons, something's probably wrong.

**Review by hugh-emerson — APPROVED** [review]
I think your solution is a bit overcomplicated and makes the tests worse if anything... see my other comments for details.

Also, general point - function and variable names should be snake_case.

But I'm marking this as complete as far as the course goes because you do have working tests.

File: src/todolist/Task.py (hunk 2 above)
Comment (JoLeeSoftwire) [reply src/todolist/Task.py]: I will revert most of this and have another go.
It's probably worth me feeding back that I was trying to apply the same restructuring as in the example on pgs 25-27 of the reading material for module 3. I've clearly misapplied it, but worth mentioning that's where I got it from.

**Pull request merged**